### PR TITLE
Introduce multiple changes to standardize and productionize repository for release

### PR DESCRIPTION
**Pull request by aiyengar2**
Needs to be paired with merging https://github.com/rancher/ui-plugin-examples/pull/6 to avoid breaking the example repository.

Breakages are primarily caused by contents of two commits:
- `Rename charts to charts/ui-plugin-server`: breaks expected directory in https://github.com/rancher/ui-plugin-examples/blob/61203833c330678819f703a351786f2d90ad08ef/scripts/publish#L171
- `Modify behavior of patch command to work with make`: breaks expected pattern for calling script in https://github.com/rancher/ui-plugin-examples/blob/61203833c330678819f703a351786f2d90ad08ef/scripts/publish#L177

### CODE REVIEW

File: package/nginx.conf
Comment (aiyengar2) [inline package/nginx.conf]: purpose of adding this here is to be able to do local testing, I'll probably add some stuff to run the docker command and curl some stuff for smoke tests to CI when I get to it

File: plugin/package.json
Comment (aiyengar2) [inline plugin/package.json]: this tests the package.json logic for running the `patch` operation that I encoded inside with more stuff than before; I used the logic in https://github.com/rancher/ui-plugin-examples/blob/main/scripts/helmgen to encode some more stuff into the scripts.

File: charts/ui-plugin-server/values.yaml
Comment (aiyengar2) [inline charts/ui-plugin-server/values.yaml]: hmm this is the detriment of using yq to process changes in the values.yaml... maybe it might make more sense to use `sed` for values.yaml operations to preserve newline formatting?

This is an open issue in yq noted in https://github.com/mikefarah/yq/issues/465.

File: charts/ui-plugin-server/Chart.yaml
Comment (aiyengar2) [inline charts/ui-plugin-server/Chart.yaml:3]: I moved the charts/ contents to charts/ui-plugin-server primarily for semantic reasons. @eliyamlevy do you think we should rename this directory to `chart/` or keep it as `charts/ui-plugin-server`?

File: charts/ui-plugin-server/Chart.yaml
Comment (aiyengar2) [inline charts/ui-plugin-server/Chart.yaml:11]: these are automatically written from the package.json, see the new functionality of the patch script

File: docs/gettingstarted.md
Comment (aiyengar2) [inline docs/gettingstarted.md:1]: I wrote this fairly quickly, so please let me know if you catch typos! Just expanded on the stuff you wrote in the README.md

File: package/Dockerfile
Comment (aiyengar2) [inline package/Dockerfile:17]: scopes down the user to minimal permissions so you do not need to run as root

File: scripts/build
Comment (aiyengar2) [inline scripts/build:4]: removed in favor of calling this `scripts/package`, since `scripts/build` generally corresponds to `go build` in Rancher repos

File: scripts/ci
Comment (aiyengar2) [inline scripts/ci:6]: Need to add testing steps here

Comment (aiyengar2) [inline scripts/ci:6]: will also need to ensure this gets called in GH workflows

File: charts/ui-plugin-server/Chart.yaml
Comment (eliyamlevy) [reply charts/ui-plugin-server/Chart.yaml:3]: we can do the rename.

**Review by eliyamlevy — APPROVED** [review]

**Comment by eliyamlevy** [comment]
Waiting on UI to approve the changes on their side.

**Review by eliyamlevy — APPROVED** [review]
LGTM!

**Comment by aiyengar2** [comment]
Once this PR has been merged, the following requirements will be met:

Plugin Server Requirements:
- [x] Must serve an HTTP server on a given port that supports two endpoints:
  - [x] /files.txt: serves up a text file that contains a list of files (not globs) that need to be pulled to fully render a plugin. The contents of this files.txt should be equivalent to running 'find <plugin-dir> -type f' on build.
  - [x] /files/*: serves up each file found in the plugin directory.
  - [x] /version.txt: serves up a text file that contains the version of this plugin
- [x] On packaging the plugin server, a base image that contains the plugin server should be uploaded to Rancher's DockerHub.
  - @eliyamlevy, do we need to set this up anymore? Afaik this is not required since UI directly handles creating images in https://github.com/rancher/ui-plugin-examples, so there's no reason to actually publish the image built in this repository since it's never intended to be deployed without some additional scripting. If not, we need this example to be packaged up and Drone and DockerHub to be set up.
  - If there's no image, how does QA test this today? Do they override it with a dev created image or do we need QA-specific docs to be added to this repo on how to build the Docker image, publish it to a fork, and test? cc: @ronhorton 
- [x] Developer documentation and CI scripting must be created
- [x] Make-based scripts should be executable to build a local Docker image that contains the plugin server
- [x] Github Workflows / Drone workflows should be enabled on submitting PRs that run basic CI or at least attempt to ensure that building the image is successfully done. Most of this should be scripted up in https://github.com/rancher/go-skel to be used to generate a new repository (although possibly out-of-date).
- [x] Repository should contain basic files like SECURITY.md, MAINTAINERS.md, CODEOWNERS, CONTRIBUTING.md, CODE_OF_CONDUCT.md, LICENSE, etc.
- [x] Docs should be included for design, developing, and getting started. See https://github.com/rancher/prometheus-federator/tree/main/docs for an example of what should be written.

**Comment by aiyengar2** [comment]
Re-requesting review now that PR is ready to be merged.

**Do not merge until https://github.com/rancher/ui-plugin-examples/pull/6 is also approved and merged.**

**Comment by aiyengar2** [comment]
To verify from QA's perspective that my changes will **not** cause any regressions, I went back to the main branch and ran `helm template -n cattle-ui-plugin-system ui-plugin-server charts > main_output.yaml` and compared its output to `helm template -n cattle-ui-plugin-system ui-plugin-server charts/ui-plugin-server > wip_output.yaml`. 

Here is the diff computed via `diff -uN main_output.yaml wip_output.yaml`

```diff
--- main_output.yaml	2022-10-03 17:13:27.000000000 -0700
+++ wip_output.yaml	2022-10-03 17:13:47.000000000 -0700
@@ -1,5 +1,5 @@
 ---
-# Source: test-plugin/templates/hardened.yaml
+# Source: ui-plugin-server/templates/hardened.yaml
 apiVersion: networking.k8s.io/v1
 kind: NetworkPolicy
 metadata:
@@ -15,14 +15,14 @@
   - Ingress
   - Egress
 ---
-# Source: test-plugin/templates/hardened.yaml
+# Source: ui-plugin-server/templates/hardened.yaml
 apiVersion: policy/v1beta1
 kind: PodSecurityPolicy
 metadata:
-  name: test-plugin-patch-sa
+  name: ui-plugin-server-patch-sa
   namespace: cattle-ui-plugin-system
   labels:
-    app: test-plugin-patch-sa
+    app: ui-plugin-server-patch-sa
 spec:
   privileged: false
   hostNetwork: false
@@ -46,14 +46,18 @@
   volumes:
     - 'secret'
 ---
-# Source: test-plugin/templates/serviceaccount.yaml
+# Source: ui-plugin-server/templates/serviceaccount.yaml
 apiVersion: policy/v1beta1
 kind: PodSecurityPolicy
 metadata:
-  name: test-plugin
+  name: ui-plugin-server
   namespace: cattle-ui-plugin-system
-  labels:
-    app: test-plugin
+  labels: 
+    helm.sh/chart: ui-plugin-server-0.1.0
+    app.kubernetes.io/name: ui-plugin-server
+    app.kubernetes.io/instance: ui-plugin-server
+    app.kubernetes.io/version: "0.1.0"
+    app.kubernetes.io/managed-by: Helm
 spec:
   privileged: false
   hostNetwork: false
@@ -77,29 +81,39 @@
   volumes:
     - 'secret'
 ---
-# Source: test-plugin/templates/hardened.yaml
+# Source: ui-plugin-server/templates/hardened.yaml
 apiVersion: v1
 kind: ServiceAccount
 metadata:
-  name: test-plugin-patch-sa
+  name: ui-plugin-server-patch-sa
   namespace: cattle-ui-plugin-system
   labels:
-    app: test-plugin-patch-sa
+    app: ui-plugin-server-patch-sa
 ---
-# Source: test-plugin/templates/serviceaccount.yaml
+# Source: ui-plugin-server/templates/serviceaccount.yaml
 apiVersion: v1
 kind: ServiceAccount
 metadata:
-  name: test-plugin
+  name: ui-plugin-server
   namespace: cattle-ui-plugin-system
-  labels:
-    app: test-plugin
+  labels: 
+    helm.sh/chart: ui-plugin-server-0.1.0
+    app.kubernetes.io/name: ui-plugin-server
+    app.kubernetes.io/instance: ui-plugin-server
+    app.kubernetes.io/version: "0.1.0"
+    app.kubernetes.io/managed-by: Helm
 ---
-# Source: test-plugin/templates/nginxconf.yaml
+# Source: ui-plugin-server/templates/nginxconf.yaml
 apiVersion: v1
 kind: ConfigMap
 metadata:
-  name: test-plugin-nginx-conf
+  name: ui-plugin-server-nginx-conf
+  labels: 
+    helm.sh/chart: ui-plugin-server-0.1.0
+    app.kubernetes.io/name: ui-plugin-server
+    app.kubernetes.io/instance: ui-plugin-server
+    app.kubernetes.io/version: "0.1.0"
+    app.kubernetes.io/managed-by: Helm
 data:
   nginx.conf: |
     events {}
@@ -120,13 +134,13 @@
         }
     }
 ---
-# Source: test-plugin/templates/hardened.yaml
+# Source: ui-plugin-server/templates/hardened.yaml
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRole
 metadata:
-  name: test-plugin-patch-sa
+  name: ui-plugin-server-patch-sa
   labels:
-    app: test-plugin-patch-sa
+    app: ui-plugin-server-patch-sa
 rules:
 - apiGroups:
   - ""
@@ -137,50 +151,54 @@
   resources: ['podsecuritypolicies']
   verbs:     ['use']
   resourceNames:
-  - test-plugin-patch-sa
+  - ui-plugin-server-patch-sa
 ---
-# Source: test-plugin/templates/hardened.yaml
+# Source: ui-plugin-server/templates/hardened.yaml
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRoleBinding
 metadata:
-  name: test-plugin-patch-sa
+  name: ui-plugin-server-patch-sa
   labels:
-    app: test-plugin-patch-sa
+    app: ui-plugin-server-patch-sa
 roleRef:
   apiGroup: rbac.authorization.k8s.io
   kind: ClusterRole
-  name: test-plugin-patch-sa
+  name: ui-plugin-server-patch-sa
 subjects:
 - kind: ServiceAccount
-  name: test-plugin-patch-sa
+  name: ui-plugin-server-patch-sa
   namespace: cattle-ui-plugin-system
 ---
-# Source: test-plugin/templates/serviceaccount.yaml
+# Source: ui-plugin-server/templates/serviceaccount.yaml
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRoleBinding
 metadata:
-  name: test-plugin
-  labels:
-    app: test-plugin
+  name: ui-plugin-server
+  labels: 
+    helm.sh/chart: ui-plugin-server-0.1.0
+    app.kubernetes.io/name: ui-plugin-server
+    app.kubernetes.io/instance: ui-plugin-server
+    app.kubernetes.io/version: "0.1.0"
+    app.kubernetes.io/managed-by: Helm
 roleRef:
   apiGroup: rbac.authorization.k8s.io
   kind: ClusterRole
-  name: test-plugin
+  name: ui-plugin-server
 subjects:
 - kind: ServiceAccount
-  name: test-plugin
+  name: ui-plugin-server
   namespace: cattle-ui-plugin-system
 ---
-# Source: test-plugin/templates/service.yaml
+# Source: ui-plugin-server/templates/service.yaml
 apiVersion: v1
 kind: Service
 metadata:
-  name: test-plugin
-  labels:
-    helm.sh/chart: test-plugin-0.1.0
-    app.kubernetes.io/name: test-plugin
+  name: ui-plugin-server
+  labels: 
+    helm.sh/chart: ui-plugin-server-0.1.0
+    app.kubernetes.io/name: ui-plugin-server
     app.kubernetes.io/instance: ui-plugin-server
-    app.kubernetes.io/version: "1.16.0"
+    app.kubernetes.io/version: "0.1.0"
     app.kubernetes.io/managed-by: Helm
 spec:
   type: ClusterIP
@@ -189,31 +207,31 @@
       targetPort: 8080
       protocol: TCP
       name: http
-  selector:
-    app.kubernetes.io/name: test-plugin
+  selector: 
+    app.kubernetes.io/name: ui-plugin-server
     app.kubernetes.io/instance: ui-plugin-server
 ---
-# Source: test-plugin/templates/deployment.yaml
+# Source: ui-plugin-server/templates/deployment.yaml
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: test-plugin
-  labels:
-    helm.sh/chart: test-plugin-0.1.0
-    app.kubernetes.io/name: test-plugin
+  name: ui-plugin-server
+  labels: 
+    helm.sh/chart: ui-plugin-server-0.1.0
+    app.kubernetes.io/name: ui-plugin-server
     app.kubernetes.io/instance: ui-plugin-server
-    app.kubernetes.io/version: "1.16.0"
+    app.kubernetes.io/version: "0.1.0"
     app.kubernetes.io/managed-by: Helm
 spec:
   replicas: 1
   selector:
-    matchLabels:
-      app.kubernetes.io/name: test-plugin
+    matchLabels: 
+      app.kubernetes.io/name: ui-plugin-server
       app.kubernetes.io/instance: ui-plugin-server
   template:
     metadata:
-      labels:
-        app.kubernetes.io/name: test-plugin
+      labels: 
+        app.kubernetes.io/name: ui-plugin-server
         app.kubernetes.io/instance: ui-plugin-server
     spec:
       nodeSelector: 
@@ -224,10 +242,10 @@
           effect: "NoSchedule"
           operator: "Equal"
       containers:
-        - name: test-plugin
-          securityContext:
+        - name: ui-plugin-server
+          securityContext: 
             {}
-          image: "yourrepohere/plugin-server:v0"
+          image: "rancher/ui-plugin-server:0.1.0"
           imagePullPolicy: Always
           ports:
             - name: http
@@ -236,56 +254,61 @@
           volumeMounts:
           - mountPath: /etc/nginx # mount nginx-conf volumn to /etc/nginx
             readOnly: true
-            name: test-plugin-nginx-conf
-          # livenessProbe:
-          #   httpGet:
-          #     path: /
-          #     port: http
-          # readinessProbe:
-          #   httpGet:
-          #     path: /
-          #     port: http
+            name: ui-plugin-server-nginx-conf
+          readinessProbe:
+            httpGet:
+              scheme: http
+              path: /
+              port: http
+              initialDelaySeconds: 10
+              periodSeconds: 5
           resources:
       volumes:
-        - name: test-plugin-nginx-conf
+        - name: ui-plugin-server-nginx-conf
           configMap:
-            name: test-plugin-nginx-conf # place ConfigMap `nginx-conf` on /etc/nginx
+            name: ui-plugin-server-nginx-conf # place ConfigMap `nginx-conf` on /etc/nginx
             items:
               - key: nginx.conf
                 path: nginx.conf
 ---
-# Source: test-plugin/templates/cr.yaml
+# Source: ui-plugin-server/templates/cr.yaml
 apiVersion: catalog.cattle.io/v1
 kind: UIPlugin
 metadata:
-  name: test-plugin
+  name: ui-plugin-server
   namespace: cattle-ui-plugin-system
+  labels: 
+    helm.sh/chart: ui-plugin-server-0.1.0
+    app.kubernetes.io/name: ui-plugin-server
+    app.kubernetes.io/instance: ui-plugin-server
+    app.kubernetes.io/version: "0.1.0"
+    app.kubernetes.io/managed-by: Helm
 spec:
   plugin: # should initially follow the design of the Helm Chart.yaml fields, could discuss modifying this
-    name: test-plugin
-    version: 0.0.1
-    endpoint: http://test-plugin.cattle-ui-plugin-system.svc
+    name: ui-plugin-server
+    version: 0.1.0
+    endpoint: http://ui-plugin-server.cattle-ui-plugin-system.svc
     noCache: false
 ---
-# Source: test-plugin/templates/hardened.yaml
+# Source: ui-plugin-server/templates/hardened.yaml
 apiVersion: batch/v1
 kind: Job
 metadata:
-  name: test-plugin-patch-sa
+  name: ui-plugin-server-patch-sa
   namespace: cattle-ui-plugin-system
   labels:
-    app: test-plugin-patch-sa
+    app: ui-plugin-server-patch-sa
   annotations:
     "helm.sh/hook": post-install, post-upgrade
     "helm.sh/hook-delete-policy": hook-succeeded, before-hook-creation
 spec:
   template:
     metadata:
-      name: test-plugin-patch-sa
+      name: ui-plugin-server-patch-sa
       labels:
-        app: test-plugin-patch-sa
+        app: ui-plugin-server-patch-sa
     spec:
-      serviceAccountName: test-plugin-patch-sa
+      serviceAccountName: ui-plugin-server-patch-sa
       securityContext:
         runAsNonRoot: true
         runAsUser: 1000
```

As is visible in the diff **the only real changes added inside are primarily naming changes and standardizations**. And since this PR introduces rigorous automated testing for the Docker image, the likelihood of regressions there is also unlikely.

**Review by eliyamlevy — APPROVED** [review]

**Review by eliyamlevy — APPROVED** [review]

**Pull request merged**
